packages/js/src/ui/context/CountContext.tsx

```diff
@@ -102,26 +105,49 @@ export const useTotalUnreadCount = () => {
   return { totalUnreadCount: context.totalUnreadCount };
 };
 
-export const useNewMessagesCount = (props: { tags: string[] }) => {
+type UseNewMessagesCountProps = {
+  filter: Pick<NotificationFilter, 'tags'>;
+};
+export const useNewMessagesCount = (props: UseNewMessagesCountProps) => {
   const context = useContext(CountContext);
   if (!context) {
     throw new Error('useNewMessagesCount must be used within a CountProvider');
   }
 
-  const key = createMemo(() => createKey(props.tags));
+  const key = createMemo(() => createKey(props.filter));
   const count = createMemo(() => context.newNotificationCounts().get(key()) || 0);
 
   return { count };
 };
 
-export const useUnreadCount = (props: { tags: string[] }) => {
+type UseUnreadCountProps = {
+  filter: Pick<NotificationFilter, 'tags'>;
```

Comment:
✏ **suggestion:** it is a little confusing as the filter is `tags + read + archived` fields... I would keep it as `tags`